include/boost/geometry/algorithms/detail/overlay/sort_by_side.hpp

```diff
@@ -316,7 +316,8 @@ public :
         // then take a point (or more) further back.
         // The limit of offset avoids theoretical infinite loops.
         // In practice it currently walks max 1 point back in all cases.
-        double const tolerance = 1.0e9;
+        using ct_type = typename geometry::coordinate_type<Point>::type;
+        ct_type const tolerance = 1000 * 1000 * 1000;
```

Comment:
I this this will not compile properly on platforms with 16-bit `int` becasue there will be an overflow. We probably don't target such platforms but still. What was the reason behind this change?

Reply:
I also not understand this change. Is this to support multiprecision? Why not `ct_type const tolerance = 1000000000` ?

Reply:
I'll fix it.

Because I'm now working in two branches which are quite related to each other, and they are both approved, I will decline this PR and move this commit to the other PR.